src/keyboard/mod.rs

```diff
@@ -0,0 +1,258 @@
+//! Utilities for keyboard handling
+//!
+//! This module provides utilities for keyboardand keymap handling: keymap interpretation
+//! and forwarding keystrokes to clients using xkbcommon.
+//!
+//! You can first create a `KbdHandle` using the `create_keyboard_handler` function in this module.
+//! The handle you obtained can be cloned to access this keyboard state from different places. It is
+//! expected that such a context is created for each keyboard the compositor has access to.
```

Comment:
Weston on its side seems to just use the seats as they are defined by libinput: https://github.com/wayland-project/weston/blob/75b7197f4e072a4e2de124ddbe93b85cffb1c0f8/libweston/libinput-seat.c#L412-L423 (the logic is quite scattered around this file actually...)

My understanding is that it keeps an xkbcommon state for each seat, and fuses all keyboard of a seat into a single logical keyboard. That's quite closer to my initial understanding of how libinput works.